File: backend/src/main/java/com/yigongil/backend/domain/certification/CertificationRepository.java
package com.yigongil.backend.domain.certification;

import com.yigongil.backend.exception.NoCertificationException;
import java.util.Optional;
import org.springframework.data.repository.Repository;

public interface CertificationRepository extends Repository<Certification, Long> {

    Certification save(Certification feedPost);

    Optional<Certification> findByRoundIdAndAuthorId(Long roundId, Long memberId);

    default Certification getByRoundIdAndAuthorId(Long roundId, Long memberId) {
        return findByRoundIdAndAuthorId(roundId, memberId).orElseThrow(() -> new NoCertificationException("인증을 찾을 수 없습니다", String.valueOf(memberId)));
    }
}

File: android/domain/src/main/java/com/created/domain/repository/CertificationRepository.kt
package com.created.domain.repository

import com.created.domain.model.Certification
import com.created.domain.model.MemberCertification
import java.io.File

interface CertificationRepository {

    suspend fun postImage(
        file: File,
    ): Result<String>

    suspend fun postCertification(
        studyId: Long,
        certification: Certification,
    )

    suspend fun getMemberCertification(
        studyId: Long,
        roundId: Long,
        memberId: Long,
    ): MemberCertification
}
